containers/ecr-viewer/src/app/api/services/core_database_repo.ts

```
@@ -0,0 +1,337 @@
+import { Kysely } from "kysely";
+
+import {
+  ECRConditions,
+  NewECRConditions,
+  ECRConditionsUpdate,
+  ECRRuleSummaries,
+  NewECRRuleSummaries,
+  ECRRuleSummariesUpdate,
+  ECR,
+  NewECR,
+  ECRUpdate,
+} from "./core_types";
+import { Core } from "./core_types";
+import { db } from "./database";
+
+// ECR_DATA
+
+/**
+ * Finds an eICR by its ID
+ * @async
+ * @function findEcrById
+ * @param id - thd ID of the eCR to be looked up
+ * @returns an eICR object
+ */
+export async function findEcrById(id: string | null): Promise<ECR | undefined> {
+  if (!id) {
+    throw new Error("eICR ID is required.");
+  }
+  try {
+    return await (db as Kysely<Core>)
+      .selectFrom("ecr_data")
+      .where("eICR_ID", "=", id)
+      .selectAll()
+      .executeTakeFirst();
+  } catch (error) {
+    console.error(error);
```

are we purposefully swallowing these errors instead of continuing to throw them?